.gitignore

```diff
@@ -1,2 +1,5 @@
 node_modules
 dist
+package-lock.json
+package-lock.json.*
```

Comment:
> but I wonder if it is necessary since the project is yarn-based so I would avoid merging these in in any case

I think that it is necessary, since as far as I know it is impossible to actually enforce yarn for other contributors. You may be using it locally, but a contributor using npm will generate a package lock file.

i.e. I typically just default to npm when contributing to open source projects, so I pulled this project down, ran `npm i`, and a package-lock file was generated for me.

> Related tangent: I've preferred yarn, but I have wondered if I should stick just to npm for libs/open source stuff

I think that you can use either for your own development (it really doesn't matter what you use, I think), but I think it is cool when the project is set up to support a developer using either.

Reply:
Yeah, good point. I will keep that in mind for future. If I'm going to favor yarn in my projects then I think I need to ignore `package-lock*`, track `yarn.lock`, and make sure `yarn.lock` gets updated anytime `package.json` dependencies are changed (I can update this myself for contributors not using `yarn`). I do like tracking a lock file but would want to avoid tracking/maintaining both.

I think I'll continue with that for now, but I I'm going to let the question of just sticking with npm stew further if it might simplify things/contributions.